src/generate/register.rs

```diff
@@ -264,7 +256,41 @@ pub fn fields(
 
         let mut evs_r = None;
 
+        let field_dim = match f {
+            Field::Array(_, de) => {
+                let (first, index) = if let Some(dim_index) = &de.dim_index {
+                    if let Ok(first) = dim_index[0].parse::<u32>() {
+                        let sequential_indexes = dim_index
+                            .iter()
+                            .map(|element| element.parse::<u32>())
+                            .eq((first..de.dim + first).map(Ok));
+                        if !sequential_indexes {
+                            return Err(format!("unsupported array indexes in {}", f.name))?;
+                        }
+                        (first, None)
+                    } else {
+                        (0, de.dim_index.clone())
+                    }
+                } else {
+                    (0, None)
+                };
+                let suffixes: Vec<_> = match index {
+                    Some(ix) => ix,
+                    None => (0..de.dim).map(|i| (first + i).to_string()).collect(),
+                };
+                let suffixes_str = format!("({}-{})", first, first + de.dim - 1);
+                Some((first, de.dim, de.dim_increment, suffixes, suffixes_str))
+            }
+            Field::Single(_) => None,
+        };
+
         if can_read {
+            let readerdoc = if let Some((_, _, _, _, suffixes_str)) = &field_dim {
+                format!("Reader of fields `{}{}`", &name, &suffixes_str)
```

Comment:
Maybe this fixes the last issue
```suggestion
                format!("Reader of fields `{}`", util::replace_suffix(&f.name, &suffixes_str))
```

Reply:
The same for "Write proxy for fields"